### PR TITLE
do not consider empty string in word matching

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/flows/tests.py [hunk 1]
```diff
@@ -1165,6 +1169,16 @@ def test_tests(self):
         sms.text = "kigali city"
         self.assertTest(False, None, test)
 
+        sms.text = "blue white, allo$%%"
+        self.assertTest(False, None, test)
+
+        test = ContainsAnyTest(test=dict(base="%%$, &&,"))
```
Comment (norkans7) [inline temba/flows/tests.py]: Probably we need frontend validation for test to not end up not having no word to match at all
or warn the user the part that we will ignore

**Comment by norkans7** [comment]
fix https://github.com/rapidpro/rapidpro/issues/324

**Comment by nicpottier** [comment]
What was the burmese text they used in their test?

**Comment by norkans7** [comment]
This is not about Burmese, just that we split out the non words character https://github.com/nyaruka/rapidpro/blob/master/temba/flows/models.py#L5566 and below and we could end up with empty strings in the list

for Burmese the split make a list of  empty strings and ? so that wrong behavior is still there we need to chat more how it can be solved

**Comment by nicpottier** [comment]
Ah ok, nice! :octocat:

**Pull request merged**